src/modules/workspaces/workspaces.usecase.spec.ts

```diff
@@ -949,4 +950,441 @@ describe('WorkspacesUsecases', () => {
+        jest
+          .spyOn(teamRepository, 'getTeamsInWorkspaceCount')
+          .mockResolvedValue(5);
+        jest.spyOn(teamRepository, 'createTeam').mockResolvedValue(createdTeam);
+
+        const newTeam = await service.createTeam(user, workspace.id, teamInput);
+
+        expect(teamRepository.createTeam).toHaveBeenCalledWith(
+          expect.objectContaining({
+            workspaceId: workspace.id,
+            name: teamInput.name,
+            managerId: user.uuid,
+          }),
+        );
+
+        expect(newTeam).toEqual(createdTeam);
+      });
+    });
+
+    describe('editTeamData', () => {
+      it('When team is not found, then it should throw', async () => {
+        const teamId = 'team-id';
+        const editTeamDto = { name: 'Updated Team' };
+
+        jest.spyOn(teamRepository, 'getTeamById').mockResolvedValue(null);
+
+        await expect(service.editTeamData(teamId, editTeamDto)).rejects.toThrow(
+          BadRequestException,
+        );
+      });
+
+      it('When team is default company of a workspace, then it should throw', async () => {
+        const user = newUser();
+        const workspace = newWorkspace({ owner: user });
+        const team = newWorkspaceTeam({
+          workspaceId: workspace.id,
+          manager: user,
+        });
+        workspace.defaultTeamId = team.id;
+        const editTeamDto = { name: 'Updated Team' };
+
+        jest.spyOn(teamRepository, 'getTeamById').mockResolvedValue(team);
+        jest.spyOn(workspaceRepository, 'findOne').mockResolvedValue(workspace);
+
+        await expect(
+          service.editTeamData(team.id, editTeamDto),
+        ).rejects.toThrow(BadRequestException);
+      });
+
+      it('When team is being updated, then it update the respective fields', async () => {
+        const user = newUser();
+        const workspace = newWorkspace({ owner: user });
+        const team = newWorkspaceTeam({
+          workspaceId: workspace.id,
+          manager: user,
+        });
+        const editTeamDto = { name: 'Updated Team' };
+
+        jest.spyOn(teamRepository, 'getTeamById').mockResolvedValue(team);
+        jest.spyOn(workspaceRepository, 'findOne').mockResolvedValue(workspace);
+
+        await service.editTeamData(team.id, editTeamDto);
+
+        expect(teamRepository.updateById).toHaveBeenCalledWith(
+          team.id,
+          editTeamDto,
+        );
+      });
+    });
+
+    describe('addMemberToTeam', () => {
+      it('When team is not found, then it should throw', async () => {
+        const teamId = 'team-id';
+        const memberId = 'member-uuid';
+
+        jest.spyOn(teamRepository, 'getTeamById').mockResolvedValue(null);
+
+        await expect(service.addMemberToTeam(teamId, memberId)).rejects.toThrow(
+          BadRequestException,
+        );
+      });
+
+      it('When team is default company of a workspace, then it should throw', async () => {
```

Comment:
_When the team to update is the default one of a workspace_ could be a better legend